odoo/addons/base/tests/test_gettext_alias.py

```diff
@@ -0,0 +1,48 @@
+import inspect
+from unittest.mock import patch
+
+from odoo.tests import common
+from odoo.tools.translate import GettextAlias
+
+
+class TestGettextAlias(common.TransactionCase):
+    @classmethod
+    def setUp(self):
+        super().setUpClass()
+
+    @patch("odoo.tools.translate.code_translations.get_python_translations")
+    @patch("odoo.modules.get_resource_from_path")
+    @patch.object(GettextAlias, "_get_lang", return_value="fr_FR")
+    def test_get_translation_with_lang_in_self(self, mock_get_lang, mock_get_resource_from_path, mock_get_python_translations):
```

Comment:
The tests seem not so intuitive to me (though it might be me 😄).
Pretty much everything is mocked and the only thing that is checked is whether the `_get_lang` function is called with the correct stackframe and the `get_python_translations` function is getting called with the right module and language (which is mocked as well).

I'd rather expect to see a test that mimics a real use-case, since we're never calling `GettextAlias._get_translation()` directly, but `_()`. Something like:
```py
    @patch("odoo.tools.translate.code_translations.get_python_translations")
    def test_get_translation_lang(self, patched_get_python_translations):
        context = {"lang": "fr_FR"}
        _("Hello")
        patched_get_python_translations.assert_called_with("base", "fr_FR")

    @patch("odoo.tools.translate.code_translations.get_python_translations")
    def test_get_translation_lang_list_comp(self, patched_get_python_translations):
        context = {"lang": "ar_001"}
        [_("Hello") for i in range(1)]
        patched_get_python_translations.assert_called_with("base", "ar_001")
```
Could be a bit more extensive by trying different ways in which the `lang` can be set, but just to get an idea.

I guess the way it is tested now kind of checks whether it works, but pretty indirectly.

Don't know what you think of this @oomsveta?